code/modules/mob/living/carbon/human/say.dm

```diff
@@ -5,22 +5,22 @@
 		verb_ask = pick(T.ask_mod)
 		verb_yell = pick(T.yell_mod)
 		verb_exclaim = pick(T.exclaim_mod)
+	if(wear_mask)
+		var/obj/item/clothing/mask/M = wear_mask
+		if(!isnull(M.chosen_tongue))
+			verb_say = pick(initial(M.chosen_tongue.say_mod))
+			verb_ask = pick(initial(M.chosen_tongue.ask_mod))
+			verb_yell = pick(initial(M.chosen_tongue.yell_mod))
+			verb_exclaim = pick(initial(M.chosen_tongue.exclaim_mod))
 	if(slurring || !T)
 		return "slurs"
 	else
 		. = ..()
 
 /mob/living/carbon/human/GetVoice()
-	if(istype(wear_mask, /obj/item/clothing/mask/chameleon))
-		var/obj/item/clothing/mask/chameleon/V = wear_mask
-		if(V.vchange && wear_id)
-			var/obj/item/card/id/idcard = wear_id.GetID()
-			if(istype(idcard) && idcard.electric)
-				return idcard.registered_name
-			else
-				return real_name
-		else
-			return real_name
+	if(wear_mask)
+		var/obj/item/clothing/mask/gas/old/modulator/V = wear_mask
+		return V.get_name(usr)
```

Comment:
This needs to work if the user is a changeling mimicing or has a special voice and the mask is turned off. I would recommend only returning if the result is null, and returning null if you don't want to change the voice

Reply:
I'll just make it do this instead
![image](https://github.com/BeeStation/BeeStation-Hornet/assets/110184118/4eff5862-8076-4c67-8bd3-ee7e41a389ae)

Reply:
It'll return the real name by default, but then each new check will override it.
First it checks for the voice change virus, then if the user is a ling and then if the user is wearing a mask that can change the voice.